azure-iot-device/azure/iot/device/iothub/config.py

```diff
@@ -0,0 +1,15 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+class ConfigProvider(object):
```

Comment:
>ConfigProvider [](start = 6, length = 14)

ConfigProvider isn't quite the right name.  "Provider" is too generic -- I don't know what it mean to provide.  How about IoTHubPipelineOptions?  
Also, don't forget that we'll need to do something similar for DPS.  At this layer, it's OK to call it IoTHubPipelineOptions, but in the pipeline, I would just call it "PipelineOptions". I think we want the freedom to add IoTHub-only options in the future, but for now, the only option we have applies to both.  So the question becomes: do we make 2 parallel option structures for the top layer or do we have a single option structure and worry about diverging the options later?

Reply:
We were trying to shy away from calling it Options since we already have Operations shortened to Ops, and the usual shortening of Options is Ops as well... so it's a bit of a naming confusion.

Reply:
How's `IoTHubPipelineConfig`?